src/huggingface_hub/fastai_utils.py

```diff
@@ -0,0 +1,337 @@
+
+    Keyword Parameters:
+        pickle_protocol (:obj:`int`, `optional`):
+            Pickle protocol passed to torch.save. Refer to pickle documentation.
+
+    """
+    # Check that fastai and fastcore versions are supported.
+    check_fastai_fastcore_versions()
+
+    # Unpacking **kwargs
+    pickle_protocol: int = kwargs.get("pickle_protocol", 2)
+
+    os.makedirs(save_directory, exist_ok=True)
+
+    # saving config
+    # if user provides config then we update it with the fastai and fastcore versions in CONFIG_TEMPLATE.
+    if config is not None:
+        if not isinstance(config, dict):
+            raise RuntimeError(
+                f"Provided config should be a dict. Got: '{type(config)}'"
+            )
+        path = os.path.join(save_directory, CONFIG_NAME)
+        with open(path, "w") as f:
+            json.dump({**config, **CONFIG_TEMPLATE}, f)
+    else:
+        path = os.path.join(save_directory, CONFIG_NAME)
+        with open(path, "w") as f:
+            json.dump(CONFIG_TEMPLATE, f)
+
+    # creating README.md if none exist
+    _create_model_card(Path(save_directory))
+
+    # saving learner
+    learner.export(
+        fname=os.path.join(save_directory, "model.pkl"), pickle_protocol=pickle_protocol
+    )
+
+
+def from_pretrained_fastai(
+    model_id,
+    **kwargs,
+):
+    """
+    Load `model_id` files from the Hub.
+
+    Parameters:
+        model_id (:obj:`str`):
+            The model id where the pickled fastai.Learner is. Example: 'espejelomar/fastai-pet-breeds-classification'.
+
+    Keyword Parameters:
+        config (:obj:`dict`, `optional`):
+            Configuration object.
+        revision (:obj:`str`, `optional`):
+            Revision at which the repo's files are downloaded. See documentation of `snapshot_download`.
+        cache_dir (:obj:`str`, `Path`, `optional`):
+            Path to cache directory.
+
+    Returns:
+        The `fastai.Learner` model in the `model_id` repo.
+    """
+    # Check that fastai and fastcore versions are supported.
+    check_fastai_fastcore_versions()
+
+    # Import `load_learner` from `fastai.learner`.
+    from fastai.learner import load_learner
+
+    # Unpack **kwargs
+    revision: str = kwargs.get("revision", None)
+    cache_dir: str = kwargs.get("cache_dir", None)
+    config: Dict = kwargs.get("config", None)
+
+    # Root is either a local filepath matching model_id or a cached snapshot
+    if not os.path.isdir(model_id):
+        storage_folder = snapshot_download(
+            repo_id=model_id, revision=revision, cache_dir=cache_dir
+        )
+    else:
+        storage_folder = model_id
+
+    # Using the pickle document in the downloaded list
+    docs = os.listdir(storage_folder)
+    for doc in docs:
+        if doc.endswith(".pkl"):
+            pickle = doc
+            break
```

Comment:
We usually push for same naming convention. Since your save models with name `model.pkl`, you can just assume the model name is `model.pkl` here as well.